src/plugins/plugin.legend.js

```diff
@@ -193,6 +193,10 @@ var Legend = Element.extend({
 		var labelOpts = me.options.labels || {};
 		var legendItems = helpers.callback(labelOpts.generateLabels, [me.chart], me) || [];
 
+		helpers.each(legendItems, function(legendItem) {
```

Comment:
I would move this after the next `if` block. Placing it here could break existing users who don't expect the array to be joined. Also, it would be more performant to join only the filtered set

Reply:
77f932c resolves this